
github.com/ofiwg/libfabric/pull/10857
github.com/ofiwg/libfabric/commit/71caa8

From 71caa804137c07a3589f69fdb7ea220c2bbe5cbf Mon Sep 17 00:00:00 2001
From: Yaakov Selkowitz <yselkowi@redhat.com>
Date: Sun, 9 Mar 2025 23:26:04 -0400
Subject: [PATCH] prox/opx: Fix build with GCC 15

GCC 15 defaults to C23, in which `static_assert` is a keyword (rather than
an alias to `_Static_assert`), and the interpretation of function
declarations without parameters changed from unspecified (as in K&R) to
`void`.


adapted to opensuse's source tree

--- prov/opx/include/rdma/opx/fi_opx.h
+++ prov/opx/include/rdma/opx/fi_opx.h
@@ -94,7 +94,7 @@
 // Useful for checking that structures are the correct size and other
 // compile-time tests. static_assert has existed since C11 so this
 // should be safe, but we have an else clause just in case.
-#if defined(static_assert)
+#if __STDC_VERSION__ >= 201112L
 #define OPX_COMPILE_TIME_ASSERT(cond, msg) static_assert(cond,msg)
 #else
 #define OPX_COMPILE_TIME_ASSERT(cond, msg) if(0){switch(0){case 0:case cond:;}}
